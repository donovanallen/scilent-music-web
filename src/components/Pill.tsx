import { Chip } from '@nextui-org/react';
import React from 'react';

import { cn } from '@/lib/utils';

type NextPillProps = {
  text: string;
  className?: string;
  variant?:
    | 'solid'
    | 'bordered'
    | 'light'
    | 'flat'
    | 'faded'
    | 'shadow'
    | 'dot';
  size?: 'sm' | 'md' | 'lg';
  radius?: 'none' | 'sm' | 'md' | 'lg' | 'full';
  disabled?: boolean;
};

const NextPill: React.FC<NextPillProps> = ({
  text,
  className,
  variant,
  disabled = false,
  size = 'md',
  radius = 'md',
}) => {
  return (
    <Chip
      size={size}
      variant={variant}
      radius={radius}
      isDisabled={disabled}
      className={cn(className)}
      classNames={{
        base: cn('z-10 bg-opacity-90', variant === 'bordered' ? 'border' : ''),
        content: cn(
          'drop-shadow shadow-black subtitle font-thin text-2xs lg:text-xs line-clamp-1',
        ),
      }}
    >
      {text}
    </Chip>
  );
};

export default NextPill;
