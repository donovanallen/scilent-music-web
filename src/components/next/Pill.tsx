import { Chip } from '@nextui-org/react';
import React from 'react';

import { cn } from '@/lib/utils';

type NextPillProps = {
  text: string;
  className?: string;
  children?: React.ReactNode;
  variant?:
    | 'solid'
    | 'bordered'
    | 'light'
    | 'flat'
    | 'faded'
    | 'shadow'
    | 'dot';
  size?: 'sm' | 'md' | 'lg';
  disabled?: boolean;
};

const NextPill: React.FC<NextPillProps> = ({
  text,
  className,
  children,
  variant,
  disabled = false,
  size,
}) => {
  return (
    <Chip
      size={size}
      variant={variant}
      radius='lg'
      isDisabled={disabled}
      className={cn(className)}
      color='primary'
      classNames={{
        base: 'z-10 bg-opacity-90',
        content:
          'drop-shadow shadow-black text-white subtitle font-thin text-2xs lg:text-xs line-clamp-1',
      }}
    >
      {text}
    </Chip>
  );
};

export default NextPill;
