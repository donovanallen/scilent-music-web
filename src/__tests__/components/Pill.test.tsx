import { render } from '@testing-library/react';
import React from 'react';

import Pill from '@/components/Pill';

describe('Pill', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Pill text='Test' />);
    expect(asFragment()).toMatchSnapshot();
  });
});
