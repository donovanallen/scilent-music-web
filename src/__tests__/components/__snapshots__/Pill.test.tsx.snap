// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill renders correctly 1`] = `
<DocumentFragment>
  <div
    class="border border-light rounded-md px-2 py-1 w-fit overflow-clip"
  >
    <p
      class="font-thin subtitle text-2xs lg:text-xs md:truncate"
    >
      Test
    </p>
  </div>
</DocumentFragment>
`;
